### PR TITLE
feat: implement agenda

### LINKED ISSUE #13: Implement multi-day agenda section
There's a partially designed agenda, but it's missing a lot of features:
* multi day design, with the ability of changing the day
* open entry to see talk details
* talk details should be in markdown form, similar with how speakers info are kept

### PR DESCRIPTION
**Pull request by alexnm**
Closes #13 

Agenda is commented out for now until we have more details from the speakers and until we have the selection from the CFP done. Meanwhile, the only visible change is the talk details shown on the speaker modal

image

If you uncomment the agenda, you can see it in action:

image

### CODE REVIEW

File: src/components/PersonCard.astro [hunk 1]
```diff
@@ -1,43 +1,39 @@
...
 ---
 
 <div class="card" data-size={variant}>
   <button
     aria-haspopup="dialog"
-    aria-controls={id}
+    aria-controls={person.id}
     class="clickable-photo"
-    onclick={`window['${id}'].showModal()`}
+    onclick={`window['${person.id}'].showModal()`}
```
Comment (dorupopacaliman) [inline src/components/PersonCard.astro:18]:
```suggestion
    onclick={`window[person.id].showModal()`}
```

File: src/content/agenda.astro [hunk 2]
```diff
@@ -1,45 +1,108 @@
...
+}
+
+const speakers = await Astro.glob<PersonalDetails>("../data/speakers/*.md");
+const speakerTalks = await Astro.glob<TalkDetails>(
+  "../data/speaker-talks/*.md"
+);
+
+const { agendaItems, title } = Astro.props as Props;
+
+const getSpeakerInfo = (speakerId: string): Speaker | null => {
```
Comment (dorupopacaliman) [inline src/content/agenda.astro:22]: Return type should be inferred here.

File: src/content/agenda.astro [hunk 3]
```diff
@@ -1,45 +1,108 @@
...
+    (s) => speakerId === getPersonIdFromFilePath(s.file)
+  );
+
+  const speakerTalk = speakerTalks.find(
+    (s) => speakerId === getPersonIdFromFilePath(s.file)
+  );
+
+  if (!speakerDetails || !speakerTalk) {
+    return null;
+  }
```
Comment (dorupopacaliman) [inline src/content/agenda.astro:33]: I believe you only need to check for `!speakerDetails` considering you're using a fallback in the markup

File: src/content/speakers.astro [hunk 4]
```diff
@@ -1,13 +1,71 @@
...
+    (s) => speakerId === getPersonIdFromFilePath(s.file)
+  );
+
+  return {
+    type: "speaker",
+    id: speakerId,
+    bio: speakerMd.frontmatter,
+    BioMdContent: speakerMd.Content,
+    talk: speakerTalk ? speakerTalk.frontmatter : null,
+    TalkMdContent: speakerTalk ? speakerTalk.Content : null,
```
Comment (dorupopacaliman) [inline src/content/speakers.astro:40]:
```suggestion
    talk: speakerTalk?.frontmatter,
    TalkMdContent: speakerTalk?.Content,
```

**Review by dorupopacaliman — APPROVED** [review]

File: src/content/agenda.astro [hunk 5]
```diff
@@ -1,45 +1,108 @@
...
+
+  return {
+    id: speakerId,
+    bio: speakerDetails.frontmatter,
+    BioMdContent: speakerDetails.Content,
+    talk: speakerTalk.frontmatter,
+    TalkMdContent: speakerTalk.Content,
```
Comment (dorupopacaliman) [inline src/content/agenda.astro:40]:
```suggestion
    talk: speakerTalk?.frontmatter,
    TalkMdContent: speakerTalk?.Content,
```

**Comment by dorupopacaliman** [comment]
Some not important comments. Looks good.

File: src/content/agenda.astro (hunk 3 above)
Comment (alexnm) [reply src/content/agenda.astro:33]: Yes indeed, but I prefer to use `null` explicitly for marking the absence of a value. The conditional chaining operator will return a bunch of `undefined` values

File: src/content/speakers.astro (hunk 4 above)
Comment (alexnm) [reply src/content/speakers.astro:40]: This is a similar example with the comment above:
 - null marks a value that is not there and won't be there in the future
 - undefined can mean an error in data, or (not the case here) async data that hasn't been defined yet

**Pull request merged**